Document max_token_header_size TCP listener configuration option (VAULT-43023) (#2882)
hashicorp/web-unified-docs: content/vault/v2.x/content/docs/configuration/listener/tcp/index.mdx

- `max_token_header_size` `(int: 8192)` – Specifies the maximum allowed size, in
  bytes, for authentication token header values passed with `X-Vault-Token` or
  `Authorization: Bearer`. This is a defence-in-depth guardrail that protects
  against memory exhaustion via oversized variable length tokens with large 

>   `authorization_details` claims.

Reviewer: ```suggestion
  `Authorization: Bearer`. The max token size is a defence-in-depth guardrail
  that protects against memory exhaustion from oversized variable length tokens
  with large `authorization_details` claims. Vault returns a `431` status code
  and error message including `authentication token exceeds maximum allowed
  header size` for requests that exceed the configured limit return.
```
Style correction: avoid "this" as a pronoun, prefer simple language

Author: > Style correction: avoid "this" as a pronoun, prefer simple language

Thanks! Applied.

  | Value | Behaviour |
  | ----- | --------- |
  | `0` or unset | Uses the default limit of 8192 bytes (8 KB). |
  | Positive integer | Sets the limit to that many bytes. |

>   | `-1` (or any negative integer) | Removes the 8 KB application-level check. Requests are still bounded by the Go stdlib `MaxHeaderBytes` backstop (~1 MB), which returns `431 (Request Header Fields Too Large)`. |

Reviewer: ```suggestion
  | `-1` (or any negative integer) | Removes the 8 KB application-level check. Vault still enforces an upper bound on requests based on the Go stdlib `MaxHeaderBytes` backstop (~1 MB) and returns `431 (Request Header Fields Too Large)` for requests that exceed that limit. |
```
Style correction: write in active voice, be explicit about who/what takes action (Vault, the reader, another program, etc)

Author: Thanks! Applied.

  Requests that exceed the configured limit return a `431` status code, 

>   with an error message including `authentication token exceeds maximum allowed header size`.

Reviewer: ```suggestion
```
Folded into paragraph above since it's a continuation of that information.

>   **When to raise this value:** opaque Vault tokens are fixed-size and well

Reviewer: This, and the cluster listener behavior below, isn't appropriate here as it goes beyond explaining the parameter itself. If we want to have a larger discussion about when it's appropriate to change the value, the nuances of the setting, and any pitfalls, we should add a new section to the doc or a separate page to discuss request limits in Vault

Author: Move to a separate section, Token header size, at the top of document.
